examples/randomized_image_method.py

```diff
@@ -0,0 +1,168 @@
+
+ax.scatter(source_loc[0], source_loc[1], source_loc[2], marker='+', c = 'k', s=40)
+ax.scatter(mic_loc[0,:],mic_loc[1,:], mic_loc[2,:])
+
+ax.set_xlabel('X')
+ax.set_ylabel('Y')
+ax.set_zlabel('Z')
+ax.set_title('Speaker and mic array configuration') 
+
+
+plt.xlim([0,4])
+plt.ylim([0,4]) 
+
+
+###############################################################
+# normal ISM with sweeping echoes
+###############################################################
+
+# sampling frequency
+fs = 44100
+
+room = pra.ShoeBox(room_size, fs, materials=pra.Material(0.1), max_order = 50)
+
+room.add_source(source_loc)
+
+room.add_microphone_array(pra.MicrophoneArray(mic_loc, fs))
+
+room.compute_rir()
+
+#plot spectrograms to check for sweeping echoes
+
+fft_size = 512  # fft size for analysis
+fft_hop = 128  # hop between analysis frame
+fft_zp = 512  # zero padding
+analysis_window = pra.hann(fft_size)
+
+print("Sweeping echo measure for ISM is :")
+for n in range(M):
+    
+    if n == 0:
+        S = stft.analysis(room.rir[n][0],  fft_size, fft_hop, win=analysis_window, zp_back=fft_zp)
+        
+        plt.figure()
+        ax = plt.gca()
+        
+        plt.imshow(
+        pra.dB(S.T),
+        extent=[0, 0.3*fs, 0, fs / 2],
+        vmin=-100,
+        vmax=0,
+        origin="lower",
+        cmap="jet"
+        )
+        ax.set_title("RIR for Mic location " + str(n) + " without random ISM")
+        ax.set_ylabel("Frequency")
+        ax.set_xlabel("Time")
+        ax.set_aspect("auto")
+        ax.axis("off")
+
+        
+        #plot RIR
+        plt.figure()
+        plt.plot(room.rir[n][0])
+        
+        
+    # measure of spectral flatness of sweeping echos
+    # higher value is desired
+    ssf = met.sweeping_echo_measure(room.rir[n][0],fs)
+    print(ssf)   
+    
+    
+##########################################################
+# Randomized ISM should reduce sweeping echoes
+# choose a maximum displacement of 5cm
+##########################################################
+
+room = pra.ShoeBox(room_size, fs, materials=pra.Material(0.1), max_order = 50, 
+                   use_rand_ism=True, max_rand_disp=0.05)
+    
+room.add_source(source_loc)
+
+room.add_microphone_array(pra.MicrophoneArray(mic_loc, fs))
+
+room.compute_rir()
+
+   
+print("Sweeping echo measure for randomized ISM is:")
+for n in range(M):
+    
+    if n == 0:
+        S = stft.analysis(room.rir[n][0],  fft_size, fft_hop, win=analysis_window, zp_back=fft_zp)
+        
+        plt.figure()
+        ax =plt.gca()
+        
+        plt.imshow(
+        pra.dB(S.T),
+        extent=[0, 0.3*fs, 0, fs / 2],
+        vmin=-100,
+        vmax=0,
+        origin="lower",
+        cmap="jet"
+        )
+        ax.set_title("RIR for Mic location " + str(n) + " with random ISM")
+        ax.set_ylabel("Frequency")
+        ax.set_xlabel("Time")
+        ax.set_aspect("auto")
+        ax.axis("off")
+     
+        
+        #plot RIR
+        plt.figure()
```

Comment:
Here I would add a title to the RIR to match it to the corresponding spectrogram. Or put both in a subplots of a single figure.